api-inference-community/docker_images/speechbrain/app/pipelines/audio_source_separation.py

```diff
@@ -22,9 +20,36 @@ def __call__(self, inputs: np.array) -> Tuple[np.array, int]:
                 Check `app.validation` if a different sample rate is required
                 or if it depends on the model
         Return:
-            A :obj:`np.array` and a :obj:`int`: The raw waveform as a numpy array, and the sampling rate as an int.
+            est_sources: np.array  
+            
+                The raw waveforms as a numpy array of size T x S, where T is number of time points and S is the number of sources, and the sampling rate as an int.
+
+            fs : int:
+                The sampling frequency for the estimated sources
         """
-        # IMPLEMENT_THIS
-        raise NotImplementedError(
-            "Please implement AudioSourceSeparationPipeline __call__ function"
-        )
+
+
+        fs_model = self.model.hparams.sample_rate
+
+        # we resample to the model's sampling frequency if needed
+        # we assume that the input is at 16kHz
+        mix = torch.from_numpy(inputs)
+        if 16000 != fs_model:
+            resamp = torchaudio.transforms.Resample(
+                orig_freq=16000, new_freq=fs_model
+            )
+            mix = resamp(mix)
+
+        # separate
+        est_sources = self.model.separate_batch(mix)
+
+        # normalize for loudness
+        est_sources = est_sources / est_sources.max(dim=1, keepdim=True)[0]
```

Comment:
Shouldn't we use the first slice of the batch here already ?

```python
# Only 1 element in the batch is present
est_sources = est_sources[0]

#normalize for loudness
est_sources = est_sources / est_sources.max(dim=0, keepdim=True)

# return estimated sources as a T(time) x S(n.sources) matrix 
return est_sources.numpy(), fs_model
```

Reply:
Sure, these changes work for me..

Reply:
I commited the suggested changes.